### PR TITLE
Code Constification

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
I try to be conservative ;)
Feel free to comment

### CODE REVIEW

File: include/openssl/asn1t.h [hunk 1]
```diff
@@ -652,13 +652,13 @@ struct ASN1_TLC_st {
...
 /* Typedefs for ASN1 function pointers */
-
+/*
 typedef ASN1_VALUE *ASN1_new_func(void);
 typedef void ASN1_free_func(ASN1_VALUE *a);
 typedef ASN1_VALUE *ASN1_d2i_func(ASN1_VALUE **a, const unsigned char **in,
                                   long length);
 typedef int ASN1_i2d_func(ASN1_VALUE *a, unsigned char **in);
-
+*/
```
Comment (kroeckx) [inline include/openssl/asn1t.h]: Why do you comment them out?

File: crypto/engine/eng_ctrl.c [hunk 2]
```diff
@@ -241,15 +241,15 @@ int ENGINE_ctrl_cmd_string(ENGINE *e, const char *cmd_name, const char *arg,
     int num, flags;
     long l;
     char *ptr;
-    if ((e == NULL) || (cmd_name == NULL)) {
+
+    if (e == NULL || cmd_name == NULL) {
```
Comment (richsalz) [inline crypto/engine/eng_ctrl.c]: no, we don't check for NULL parameters.

File: include/openssl/asn1t.h (hunk 1 above)
Comment (richsalz) [inline include/openssl/asn1t.h]: yes, please back this change out from this MR.

File: ssl/t1_trce.c [hunk 3]
```diff
@@ -728,7 +728,7 @@ static int ssl_print_extension(BIO *bio, int indent, int server, int extype,
         break;
 
     default:
-        BIO_dump_indent(bio, (char *)ext, extlen, indent + 2);
+        BIO_dump_indent(bio, (const char *)ext, extlen, indent + 2);
```
Comment (richsalz) [inline ssl/t1_trce.c]: is this  case necessary?  it does'nt just "upgrade" the parameter automatically?

File: ssl/ssl_txt.c [hunk 4]
```diff
@@ -128,7 +128,7 @@ int SSL_SESSION_print(BIO *bp, const SSL_SESSION *x)
     if (x->tlsext_tick) {
         if (BIO_puts(bp, "\n    TLS session ticket:\n") <= 0)
             goto err;
-        if (BIO_dump_indent(bp, (char *)x->tlsext_tick, x->tlsext_ticklen, 4)
+        if (BIO_dump_indent(bp, (const char *)x->tlsext_tick, x->tlsext_ticklen, 4)
```
Comment (richsalz) [inline ssl/ssl_txt.c]: see comment below.  and other places, assume same comment :)

File: crypto/asn1/tasn_prn.c [hunk 5]
```diff
@@ -420,7 +420,7 @@ static int asn1_print_obstring(BIO *out, ASN1_STRING *str, int indent)
     } else if (BIO_puts(out, "\n") <= 0)
         return 0;
     if ((str->length > 0)
-        && BIO_dump_indent(out, (char *)str->data, str->length,
+        && BIO_dump_indent(out, (const char *)str->data, str->length,
```
Comment (richsalz) [inline crypto/asn1/tasn_prn.c]: see comment below; const not needed?

File: crypto/engine/eng_cnf.c [hunk 6]
```diff
@@ -14,9 +14,9 @@
 
 /* ENGINE config module */
 
-static char *skip_dot(char *name)
+static const char *skip_dot(const char *name)
 {
-    char *p;
+    const char *p;
```
Comment (richsalz) [inline crypto/engine/eng_cnf.c]: merge in the assignment, add a blank line.  use p != NULL below.  and perhaps the ?: operator.

File: crypto/engine/eng_ctrl.c (hunk 2 above)
Comment (richsalz) [inline crypto/engine/eng_ctrl.c]: (we're moving away from it, so at least please don't add more)

File: include/openssl/asn1t.h (hunk 1 above)
Comment (FdaSilvaYY) [inline include/openssl/asn1t.h]: Yes, it is out-of-topic., my mistake.

File: crypto/engine/eng_ctrl.c (hunk 2 above)
Comment (FdaSilvaYY) [inline crypto/engine/eng_ctrl.c]: I hadn't add any NULL check. It was just a nit.

File: ssl/t1_trce.c (hunk 3 above)
Comment (FdaSilvaYY) [inline ssl/t1_trce.c]: Yes, there was an implicit cast, that now is stated explicitly.

File: crypto/engine/eng_ctrl.c (hunk 2 above)
Comment (richsalz) [inline crypto/engine/eng_ctrl.c]: I missed the paren fixes. Fine.

File: ssl/t1_trce.c (hunk 3 above)
Comment (richsalz) [inline ssl/t1_trce.c]: Okay, I see it’s unsigned->char, you need the cast, so making it complete is fine.

**Comment by richsalz** [comment]
+1; needs a second review.

**Comment by mattcaswell** [comment]
+1

**Comment by richsalz** [comment]
merged to master, thanks!

**Comment by kaduk** [comment]
(looks like some trailing whitespace snuck in in a couple of these commits, but "meh")

**Comment by richsalz** [comment]
Yup.  Meh.

--
Senior Architect, Akamai Technologies
IM: richsalz@jabber.at Twitter: RichSalz

**Pull request closed without merging**